src/equations/numerical_fluxes.jl

```diff
@@ -37,6 +37,36 @@ end
 Base.show(io::IO, f::FluxPlusDissipation) = print(io, "FluxPlusDissipation(",  f.numerical_flux, ", ", f.dissipation, ")")
 
 
+"""
+    FluxRotated(numerical_flux)
+
+Compute a `numerical_flux` flux in direction of a normal vector by rotating the solution, 
+computing the numerical flux in x-direction, and rotating the calculated flux back.
+
+Requires a rotationally invariant equation with equation specific functions `rotate_to_x` and `rotate_from_x`.
+"""
+struct FluxRotated{NumericalFlux}
+  numerical_flux::NumericalFlux
+end
+
+@inline function (flux_rotated::FluxRotated)(u_ll, u_rr, normal_vector::AbstractVector, equations)
+  @unpack numerical_flux = flux_rotated
+
+  norm_ = norm(normal_vector)
+  # normalize(normal_vector) allocates
+  normal = SVector(normal_vector[1] / norm_, normal_vector[2] / norm_)
```

Comment:
```suggestion
  # normalize the vector without using `normalize` since we need to multiply by the `norm_` later
  normal = normal_vector / norm_
```
I did some small experiments and couldn't observe allocations using `normalize`. I would be interested to see where you observed the allocations, but that's not really important right now (since you multiply by the norm later again).

Reply:
We don't observe any allocations now either. Maybe this was due to using some Julia release candidate.